spyderlib/restart_app.py

```diff
@@ -129,18 +180,60 @@ def main():
     shell = os.name != 'nt'
 
     # Wait for original process to end before launching the new instance
-    while True:
+    max_counter = 1500  # Max loops to perform before aborting
+    counter = 0
+    while counter < max_counter:  # This is equivalent to ~ 5 minute (1500*0.2)
         if not is_pid_running(pid):
             break
         time.sleep(0.2)  # Throttling control
+        counter += 1
+
+    if counter == max_counter:

```

Comment:
With a for loop you can just use `else` here. The `else` part is run if the loop is not broken.